src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -51,13 +51,18 @@ public static class ConversionsExtensionsCatalog
         /// </summary>
         /// <remarks>This transform can operate over several columns.</remarks>
         /// <param name="catalog">The transform's catalog.</param>
-        /// <param name="columns">The input and output columns.
+        /// <param name="options">Advanced options for the estimator that also contain the input and output column names.
         /// This estimator operates over text, numeric, boolean, key and <see cref="DataViewRowId"/> data types.
         /// The new column's data type will be a vector of <see cref="System.UInt32"/>, or a <see cref="System.UInt32"/> based on whether the input column data types
         /// are vectors or scalars.</param>
-        [BestFriend]
-        internal static HashingEstimator Hash(this TransformsCatalog.ConversionTransforms catalog, params HashingEstimator.ColumnOptions[] columns)
-            => new HashingEstimator(CatalogUtils.GetEnvironment(catalog), columns);
+        ///  <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[Hash](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/HashWithOptions.cs)]
+        /// ]]></format>
+        /// </example>
+        public static HashingEstimator Hash(this TransformsCatalog.ConversionTransforms catalog, params HashingEstimator.Options[] options)
```

Comment:
>options [](start = 131, length = 7)

Wouldn't it be simpler to add the missing options `Seed` and `UseOrderedHashing` as parameters to the API above?
From looking over the transforms, I didn't see any that expose a ColumnOptions/Options class (I may have missed some though). For estimators that are trainable sometimes we expose an API for defining multiple columns using an array of `InputOutputColumnPair`, but the other options are usually common for all the columns.
Was there a specific ask for defining multiple columns with different options in a single estimator? (the hashing estimator is trainable in case of `maximumNumberOfInverts > 0`, so there is an advantage to exposing an API for multiple columns, I'm just not sure whether we need to support defining different options for each column).

Reply:
We follow the pattern as for each component there is a convenience API and an advanced API that takes in all the options.

Reply:
@eerhardt what are your thoughts?

Reply:
See the discussion on https://github.com/dotnet/machinelearning/issues/1798.

Reply:
Also see the discussion on https://github.com/dotnet/machinelearning/issues/2884. Renaming `ColumnOptions` to `Options` and leaving it as-is may not be the best choice here. See Artidoro, Senja, and Tom's comments.